packages/ui/components/localize/src/date/parseDate.js

```diff
@@ -63,12 +64,16 @@ export function parseDate(dateString, options) {
   }
 
   const [year, month, day] = parsedString.split('/').map(Number);
+  const [yearString] = parsedString.split('/');
+  if (!isCorrectYearFormat(yearString)) {
+    return undefined;
+  }
   let correctedYear = year;
   if (year < 50) {
```

Comment:
It seems we explicitly allowed this use case here? Will it still be able to reach this code or has it determined incorrect by now?